R/calcPastureSuit.R

```diff
@@ -11,58 +14,54 @@
-
-  precipitation <- list()
-  scenarios <- c("ssp126", "ssp245", "ssp370", "ssp460", "ssp585") # Current ISIMIP3bv2 scenarios
-  for (scenario in scenarios) {
-    subtype <- paste(x$version, x$climatemodel, scenario, x$period, "pr", "annual_mean", sep = ":")
-    precipitation[[scenario]] <- setNames(calcOutput("GCMClimate", subtype = subtype,
-                                                     smooth = smoothPrecipitation, aggregate = FALSE), scenario)
-  }
-  precipitation <- collapseNames(mbind(precipitation))
-
-  evapotranspiration <- calcOutput("Evapotranspiration", subtype = "H08:mri-esm2-0", aggregate = FALSE)
-
-  # temporary mapping of evapotranspiration RCP scenarios unavailable in ISIMIP3bv2
-  evapotranspiration <- add_columns(evapotranspiration, addnm = "ssp245", dim = 3.1, fill = NA)
-  evapotranspiration[, , "ssp245"] <- evapotranspiration[, , "ssp370"]
-  evapotranspiration <- add_columns(evapotranspiration, addnm = "ssp460", dim = 3.1, fill = NA)
-  evapotranspiration[, , "ssp460"] <- evapotranspiration[, , "ssp370"]
-
-  evapotranspiration <- evapotranspiration[, , getItems(precipitation, dim = 3)]
-
-  # matching available ssps scenarios
-  regex <- paste0("[", paste0("+", strtrim(getItems(evapotranspiration, dim = 3), 4), collapse = "|"), "]", "{4}$")
-  avlSSPs <- grep(regex, getNames(population), ignore.case = TRUE)
-  population <- population[, , avlSSPs]
+                           harmonize_until = 2015,
+                           aggregate = FALSE)[, , toupper(substring(x$scenario, first = 0, last = 4))]
+
+  cellPrep   <- calcOutput("LPJmLClimateInput", climatetype  = climatetype,
+                             variable     = "precipitation:monthlySum",
+                             stage        = "smoothed",
+                             lpjmlVersion = lpjml,
+                             aggregate    = FALSE)
+
+
+  cellPet    <- calcOutput(type = "LPJmL_new", climatetype = climatetype,
+                             subtype   = "mpet",
+                             stage     = "smoothed",
+                             version   = lpjml,
+                             aggregate = FALSE)
+  cellPrep <- mrwater::toolLPJcell2MAgPIEcell(cellPrep)
+  cellPet <- mrwater::toolLPJcell2MAgPIEcell(cellPet)
+
+  yearsCellPet      <- intersect(getYears(cellPet), findset("time"))
+  yearsCellPrep     <- intersect(findset("time"), getYears(cellPrep))
+  years              <- intersect(yearsCellPet, yearsCellPrep)
+  cellPrep           <- dimSums(cellPrep[, years, ], dim = 3)
+  cellPet            <- dimSums(cellPet[, years, ], dim = 3)
 
   # Cell area calculation
   landcoords <- as.data.frame(toolGetMapping("magpie_coord.rda", type = "cell", where = "mappingfolder"))
   landcoords <- cbind(landcoords, rep(1, nrow(landcoords)))
   landcoords <- raster::rasterFromXYZ(landcoords)
-  crs(landcoords) <- "+proj=longlat" # outputs cell are in km2
+  raster::crs(landcoords) <- "+proj=longlat" # outputs cell are in km2
```

Comment:
Do you really need cellsize or landarea size of a given cell? 
If you need this to calculate out of total numbers density - maybe landarea size per cell makes more sense?

there is a function already calculating this:
calcOutput("CellArea", aggregate = FALSE)